src/OrchardCore/OrchardCore.Apis.GraphQL.Abstractions/Extensions/PermissionsExtensions.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using GraphQL.Builders;
+using GraphQL.Types;
+using OrchardCore.Security.Permissions;
+
+namespace OrchardCore.Apis.GraphQL
+{
+    public static class PermissionsExtensions
+    {
+        public static readonly string PermissionsKey = "Permissions";
+
+        public static void RequirePermission(this IProvideMetadata type, Permission permission, object resource = null)
+        {
+            var permissions = type.GetMetadata<List<Tuple<Permission, object>>>(PermissionsKey);
+
+            if (permissions is null)
```

Comment:
can this ever be null? its an enumerable so I dont think this check is needed

Reply:
Side node ( is null = nice syntax )

Reply:
Should you not just use `if (permissions == null)` like @sebastienros asked?

Reply:
yes it can be null, metadata is just a dictionary keyed by a key chosen by us, we need to add it if it doesn't exist.